### PR TITLE
Use PPPort.pm for pod

### PR DESCRIPTION
**Pull request by atoomic**
GH #138

This is a suggestion to fix @khwilliamson  concern about the doc not being the correct one on metacpan.

I'm open to better suggestions

### CODE REVIEW

**Comment by pali** [comment]
I have not caught it and it looks like a big hack. So what is the problem which you are trying to fix?

**Comment by atoomic** [comment]
@pali the goal is to use the fully generated documentation on metacpan: 
https://metacpan.org/pod/Devel::PPPort

I'm playing with the idea to use a special file that I've named `Abstract.pm` which would be a generated version of `PPPort.pm` from `PPPort_pm.PL`, then include it in the dist tarball.

An alternate solution would be to update `PPPort_pm.PL` on `make dist`
I'm open to alternate solutions

thanks
nicolas

**Comment by atoomic** [comment]
@pali / @khwilliamson / @karenetheridge any positive or negative feedback on that hackery there?
maybe the alternate solution I mentioned above is a better way to go?

**Comment by pali** [comment]
I will look at this problem, but looks like it is somehow more complicated and I would need more time for it.

**Comment by khwilliamson** [comment]
I think it better not have a spurious file, but I defer to your judgment.

**Comment by karenetheridge** [comment]
There is no need for an extra file.  Just use `ABSTRACT => 'Perl/Pollution/Portability'` in Makefile.PL. To be clear: this is the string used for metadata, here: https://metacpan.org/source/ATOOMIC/Devel-PPPort-3.54/META.json#L2 ..and is what metacpan uses to display next to the module name on the release page.

**Comment by pali** [comment]
atoomic commented:
> @pali the goal is to use the fully generated documentation on metacpan:
> https://metacpan.org/pod/Devel::PPPort
> 
> _notice the **PROVIDED_API** _
> 
> I'm playing with the idea to use a special file that I've named `Abstract.pm` which would be a generated version of `PPPort.pm` from `PPPort_pm.PL`, then include it in the dist tarball.
> 
> An alternate solution would be to update `PPPort_pm.PL` on `make dist`
> I'm open to alternate solutions
> 
> thanks
> nicolas

Thanks! Now I understood the main problem.

karenetheridge commented:
> There is no need for an extra file. Just use `ABSTRACT => 'Perl/Pollution/Portability'` in Makefile.PL. To be clear: this is the string used for metadata, here: https://metacpan.org/source/ATOOMIC/Devel-PPPort-3.54/META.json#L2 ..and is what metacpan uses to display next to the module name on the release page.

@karenetheridge problem is not in `META.json`, but in the fact that `PPPort.pm` is generated on the fly and metacpan webpage https://metacpan.org/pod/Devel::PPPort does not generate it correctly. It has nothing with `ExtUtils::MakeMaker` or with with its `ABSTRACT` property.


How to solve it? I would suggest to pre-generate `PPPort.pm` in dist tarball. metacpan should take it and in build process this `PPPort.pm` can be generated (if needed) as before.

Adding `Abstract.pm` is just another hack which just complicate things and would lead to another problems. Pre-generating `PPPort.pm` is the easiest solution and you can also check before releasing that is generated correctly -- so to ensure that metacpan will show it correctly.

**Comment by atoomic** [comment]
Thanks @karenetheridge and @pali for the feedbacks
I think c223d14969 should address both concerns
and display our auto generated doc on metacpan.

File: MANIFEST.SKIP [hunk 1]
```diff
@@ -17,7 +17,6 @@
 ^parts/base-
 ^ppport\.h$
 ^PPPort\.c$
-^PPPort.pm$
```
Comment (atoomic) [inline MANIFEST.SKIP]: we are going to ship one generated `PPPort.pm` file in the tarball so metacpan could use it for documentation

File: Makefile.PL [hunk 2]
```diff
@@ -66,7 +66,7 @@ sub configure
 
   if (eval { ExtUtils::MakeMaker->VERSION(6) }) {
     push @moreopts, AUTHOR => 'Marcus Holland-Moritz <mhx@cpan.org>';
-    push @moreopts, ABSTRACT_FROM => 'PPPort_pm.PL';
+    push @moreopts, ABSTRACT => 'Perl/Pollution/Portability';
```
Comment (atoomic) [inline Makefile.PL]: just provide the accurate abstract

File: Makefile.PL [hunk 3]
```diff
@@ -88,7 +88,7 @@ sub configure
       'meta-spec' => { version => 2 },
       provides => {
         'Devel::PPPort' => {
-          file    => 'PPPort_pm.PL',
+          file    => 'PPPort.pm',
```
Comment (atoomic) [inline Makefile.PL]: ask metacpan to read `PPPort.pm` for the doc

File: Makefile.PL [hunk 4]
```diff
@@ -206,7 +206,7 @@ sub MY::dist_core
   my $rule;
   foreach $rule ( @rules ) {
     if ( $rule =~ m{^\s*^dist\s+:}m ) {
-        $rule =~ s{:}{: manifest}; # make sure we regenerate the manifest
+        $rule =~ s{:}{: PPPort.pm manifest}; # make sure we update PPPort.pm
```
Comment (atoomic) [inline Makefile.PL]: make sure we generate `PPPort.pm` and the MANIFEST on `list`

**Comment by atoomic** [comment]
@khwilliamson / @karenetheridge / @pali 
Any feedbacks on the last proposal for adjusting the online doc?
I reintroduced a patched version of `PPPort.pm` on dist that should be used for doc.

thanks
nicolas

**Comment by khwilliamson** [comment]
Whatever works

**Comment by karenetheridge** [comment]
I understand the problem now! \o/

Your latest changes look great.

**Pull request merged**